src/providers/solana/cachedRpcFactory.ts

```diff
@@ -60,67 +95,125 @@ export class CachedSolanaRpcFactory extends SolanaClusterRpcFactory {
+            .getSignatureStatuses([params[0]], {
+              searchTransactionHistory: true,
+            })
+            .send();
+          if (getSignatureStatusesResponse.value[0]?.confirmationStatus !== "finalized") {
+            return this.retryTransport<TResponse>(...args);
+          }
+        } catch (error) {
+          return this.retryTransport<TResponse>(...args);
+        }
+        break;
+      case "getBlockTime":
+        if (!this.isGetBlockTimeParams(params)) return this.retryTransport<TResponse>(...args);
+        break;
     }
 
-    const getTransactionResponse = await this.retryTransport<TResponse>(...args);
+    const response = await this.retryTransport<TResponse>(...args);
 
     // Do not cache JSON-RPC error responses, let them pass through for the RPC client to handle.
     try {
-      getThrowSolanaErrorResponseTransformer()(getTransactionResponse, { methodName: method, params });
+      getThrowSolanaErrorResponseTransformer()(response, { methodName: method, params });
     } catch {
-      return getTransactionResponse;
+      return response;
     }
 
     // Cache the transaction JSON-RPC response as we checked the transaction is finalized and not an error.
     const redisKey = this.buildRedisKey(method, params);
-    await this.redisClient?.set(
-      redisKey,
-      JSON.stringify(getTransactionResponse, jsonReplacerWithBigInts),
-      Number.POSITIVE_INFINITY
-    );
+    // Apply a random margin to spread expiry over a larger time window.
+    const standardTtl = this.baseTTL + Math.ceil(random(-ttl_modifier, ttl_modifier, true) * this.baseTTL);
+    const ttl = cacheType === CacheType.WITH_TTL ? standardTtl : Number.POSITIVE_INFINITY;
+    await this.redisClient?.set(redisKey, JSON.stringify(response, jsonReplacerWithBigInts), ttl);
 
-    return getTransactionResponse;
+    return response;
   }
 
   private buildRedisKey(method: string, params?: unknown[]) {
     // Only handles getTransaction right now.
     switch (method) {
       case "getTransaction":
         return this.getTransactionCachePrefix + JSON.stringify(params, jsonReplacerWithBigInts);
+      case "getBlockTime":
+        return this.getBlockTimeCachePrefix + JSON.stringify(params, jsonReplacerWithBigInts);
       default:
         throw new Error(`CachedSolanaRpcFactory::buildRedisKey: invalid JSON-RPC method ${method}`);
     }
   }
 
-  private cacheType(method: string): CacheType {
-    // Today, we only cache getTransaction.
-    if (method === "getTransaction") {
+  private cacheType(
+    method: string,
+    params: unknown[] = [],
+    latestFinalizedSlot = 0,
+    latestConfirmedSlot = 0
+  ): CacheType {
+    if (method === "getBlockTime") {
+      const targetSlot = (params as Parameters<GetBlockTimeApi["getBlockTime"]>)[0];
+      if (targetSlot <= latestFinalizedSlot) {
+        return CacheType.NO_TTL;
+      } else if (targetSlot > latestFinalizedSlot && targetSlot <= latestConfirmedSlot) {
```

Comment:
Loading both `finalized` and `confirmed` might be overkill and we can reduce the RPC usage by only checking one of these and making a CacheType decision off that.

For example, only fetching the `finalized` block and storing all blocks < that block with NO_TTL and all blocks above that with NONE